Document built-in plural rules in Localization using spreadsheets (#12049)
godotengine/godot-docs: tutorials/i18n/plural_rules.rst

.. _doc_plural_rules:

Plural rules
~~~~~~~~~~~~

> .. Note: This list is synced with core/string/locales.h in the engine.

Reviewer: ```suggestion
.. Note: This list should be synced with core/string/locales.h in the engine.
```
I'd say, as I don't see any automation added here

> This is the full list of built-in plural rules in Godot in human-readable form.

Reviewer: ```suggestion
This is the full list of built-in plural rules available in Godot in human-readable form.
```
I'd say

See the `source code <https://github.com/godotengine/godot/blob/4.6.3-stable/core/string/locales.h#L1819-L1855>`__

> for their original form in gettext format.

Reviewer: ```suggestion
for their original gettext format form.
```
Flows better IMO

These plural rules are automatically used when no plural rule is specified in
:ref:`CSV localization <doc_localization_using_spreadsheets>`. Plural forms are

> spelled out in order they should be specified in the CSV file (from top to

Reviewer: ```suggestion
spelled out in the order they should be specified in the CSV file (from top to
```

    :ref:`locale code <doc_locales>`.

Plural rules
------------

- Languages with no plural forms (``bm bo dz hnj id ig ii in ja jbo jv jw kde kea km ko lkt lo ms my nqo osa root sah ses sg su th to tpi vi wo yo yue zh``):

  - Only one form is used for all numbers.

- Languages with the same rules as English (``af an asa ast az bal bem bez bg brx ca ce cgg chr ckb da de dv ee el en eo es et eu fi fo fur fy gl gsw ha haw he hu ia io it jgo ji jmc ka kaj kcg kk kkj kl ks ksb ku ky lb lg lij mas mgo ml mn mr nah nb nd ne nl nn nnh no nr ny nyn om or os pap ps pt rm rof rwk saq sc sd sdh seh sn so sq ss ssy st sv sw syr ta te teo tig tk tn tr ts ug ur uz ve vo vun wae xh xog yi``):

  - Singular for 1.
  - Plural for 0, and for 2 and above.

- Languages with the same rules as French (``af an asa ast az bal bem bez bg brx ca ce cgg chr ckb da de dv ee el en eo es et eu fi fo fur fy gl gsw ha haw he hu ia io it jgo ji jmc ka kaj kcg kk kkj kl ks ksb ku ky lb lg lij mas mgo ml mn mr nah nb nd ne nl nn nnh no nr ny nyn om or os pap ps pt rm rof rwk saq sc sd sdh seh sn so sq ss ssy st sv sw syr ta te teo tig tk tn tr ts ug ur uz ve vo vun wae xh xog yi``):

  - Singular for 0 and 1.

>   - Plural for 2 and above.

Reviewer: These don't look right, the lists are:
```cpp
	{ "af an asa ast az bal bem bez bg brx ca ce cgg chr ckb da de dv ee el en eo es et eu fi fo fur fy gl gsw ha haw he hu ia io it jgo ji jmc ka kaj kcg kk kkj kl ks ksb ku ky lb lg lij mas mgo ml mn mr nah nb nd ne nl nn nnh no nr ny nyn om or os pap ps pt rm rof rwk saq sc sd sdh seh sn so sq ss ssy st sv sw syr ta te teo tig tk tn tr ts ug ur uz ve vo vun wae xh xog yi", "nplurals=2; plural=(n != 1);" },
	{ "ak bho csw ff fr guw hy kab ln mg nso pa pt_BR si ti wa", "nplurals=2; plural=(n > 1);" },
```
And are duplicated here between both lists

- Icelandic, Macedonian (``is mk``):

  - Singular when the last digit is 1 but the number is not 11.
  - Plural for all other numbers.

- Central Atlas Tamazight (``tzm``):

  - Singular for 0, 1, and all numbers from 11 to 99.
  - Plural for 2-10, and for 100 and above.

- Amharic, Assamese, Bengali, Dogri, Persian, Gujarati, Hindi, Kannada, Nigerian Pidgin, Zulu (``am as bn doi fa gu hi kn pcm zu``):

  - Singular for 2 and above.

>   - Plural for 0 and 1.

Author: This one seems odd to me (it seems swapped around at a glance), but that's what the built-in locales file contains: https://github.com/godotengine/godot/blob/35e80b3a8822a9df9be390814b62f44c0a9c69e8/core/string/locales.h#L1829

@bruvzg Can you confirm this is correct? The [original page](https://hoodie-cat.dev/godot-pluralization/) mentions it the other way around, but I changed it to match the source file.

Reviewer 2: Not sure what the ordering of the forms is, but I would use `First form` / `Second form` instead.

Reviewer 2: Or using "zero, one, few, many, other" as form names as it is labeled in https://www.unicode.org/cldr/charts/48/supplemental/language_plural_rules.html

Author: I went with the first form/second form suggestion for simplicity.
